### PR TITLE
found_trades and found_trades_id databases finished

### PR DESCRIPTION
**Pull request by adiraju13**
found_trades is a database that handles when a loop is found and stores it. Read the comments above the functions to see what each one does, and let me know if they make sense/need clarification/we need more functions.

found_trades_id is the database which keeps track of the next loop_id that can be put into the database.

All functions have been tested.

### CODE REVIEW

File: models/found_trades.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+ * Purpose: Remove edges from the graph where the owned_book is a node of an incoming or outgoing edge
+ * Inputs: user: user_id of the node book_id: book that the user owns, and a callback function
+ * Outputs: returns a callback function with error code (or success) as a parameter
+ */
+exports.remove_owned_book = function(user, owned_book, next){
+    pg.connect(process.env.DATABASE_URL, function(err, client, done){
+        done();
+        if (err){
+            console.error("Error connection to client while querying graph_edges table: ", err);
+            return next(error_codes.owned_books_errors.DB_CONNECTION_ERROR);
```
Comment (jayendrajog) [inline models/found_trades.js]: should be graph_edges_errors?

File: models/found_trades_id.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        client.query("SELECT COUNT(index) FROM found_trades_id",
+            [user_id, owned_book, target_user, wanted_book, loop_id], function(err, result){
+                if (err){
+                    console.error("Error querying table found_trades_id", err);
+                    return next(error_codes.found_trades_id_errors.DB_QUERY_ERROR);
+                }
+
+                //if relationship exists, it update it, if it doesn't, return error
+                if(result.rows[0].count == 0){
+                    client.query("INSERT INTO found_trades (index, trade_id) VALUES ($1::INTEGER, $2::INTEGER)", [0, loop_id], function(err, result){
```
Comment (jayendrajog) [inline models/found_trades_id.js]: is index supposed to be 0 every time?

**Review by jayendrajog — APPROVED** [review]
looks good to me overall

File: models/found_trades_id.js (hunk 2 above)
Comment (jerrylinew) [reply models/found_trades_id.js]: yeah, since this is just querying the count

**Pull request merged**